src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -271,13 +273,27 @@ export const handleGetPublicForm: ControllerHandler<
     case AuthType.SP:
     case AuthType.CP:
       return SpcpFactory.extractJwtPayloadFromRequest(authType, req.cookies)
-        .map(({ userName }) =>
-          res.json({
+        .map(({ userName, exp, iat, rememberMe }) => {
+          if (!exp) {
```

Comment:
why would `exp` not be defined? is it not consistent?

Reply:
and if it is consistent should we account for it in the `isSingpassJwtPayload` and `isCorppassJwtPayload` type guards?

Reply:
thanks, I cleaned up the types. exp is defined.

Reply:
Actually on review, exp is defined when the payload is received from the client because it is set with the cookie. When the payload is first generated by the server, exp is undefined. So I think it should be correct to indicate in the types that exp is optional, and to include the defensive check - see if ok?

Reply:
>when the payload is received from the client

sorry what dyou mean by this? it's always the server which determines the cookie payload right?

if `exp` only appears after we do `res.cookie`, then it seems what's going on here is we have two different types, one pre-`res.cookie` and one post-`res.cookie`. if that's the case then we should separate the types, rather than artificially combining them through an optional key (which in reality is mandatory in one and absent in the other).

Reply:
yes, I meant that `exp` and `iat` are only set upon `res.cookie`. separated the types